File: app/src/main/java/com/example/myapplication/Adapter/CatalogAdapter.java
package com.example.myapplication.Adapter;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapplication.R;
import com.squareup.picasso.Picasso;

import java.io.InputStream;
import java.net.URL;
import java.util.ArrayList;

import Models.linked.Product;
public class CatalogAdapter extends RecyclerView.Adapter<CatalogAdapter.MyView> {
    private ArrayList<Product> Products;
    public class MyView extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView CatalogItemName, CatalogItemPrice;
        ImageView CatalogPicture;
        public MyView(View view)
        {
            super(view);
            view.setOnClickListener(this);
            // initialise TextView with id
            CatalogItemName =(TextView) view.findViewById(R.id.CatalogItemName);
            CatalogPicture=(ImageView) view.findViewById(R.id.CatalogPicture);
            CatalogItemPrice=(TextView) view.findViewById(R.id.CatalogItemPrice);
        }
        @Override
        public void onClick(View view) {
            Product item = Products.get(getAdapterPosition());
            Bundle bundle = new Bundle();
            bundle.putString("id",String.valueOf(item.getProductId()));
            try {
                Navigation.findNavController(view).navigate(R.id.productFragment, bundle);
            }catch (Exception e)
            {
                e.printStackTrace();
            }
        }
    }
    public CatalogAdapter(ArrayList<Product> horizontalList)
    {
        Products = horizontalList;
    }
    public void updateUserList(final ArrayList<Product> userArrayList) {
        this.Products.clear();
        this.Products = userArrayList;
        notifyDataSetChanged();
    }
    @Override
    public MyView onCreateViewHolder(ViewGroup parent, int viewType)
    {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.catalogitem, parent, false);
        return new MyView(itemView);
    }
    @Override
    public void onBindViewHolder(final MyView holder, final int position)
    {
        try {
            holder.CatalogItemName.setText(Products.get(position).getName());
            String price=  String.valueOf(Products.get(position).getPrice());
            if (Products.get(position).getCount()<=0)
            {
                holder.CatalogItemPrice.setText(R.string.NotInStock);
            }
            else holder.CatalogItemPrice.setText(price+"P");
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
        String url = Products.get(position).getPhoto();
        try{
            Picasso.get() // Context
                    .load("http://192.168.1.180:82"+url) // URL or file
                    .into(holder.CatalogPicture);
        }
        catch (Exception e) {
            e.printStackTrace();
        }
    }
    @Override
    public int getItemCount()
    {
        return Products.size();
    }
}

File: app/src/main/java/com/example/myapplication/Adapter/CatalogAdapter.kt
package com.example.myapplication.Adapter

import Models.linked.Product
import androidx.recyclerview.widget.RecyclerView
import android.widget.TextView
import android.os.Bundle
import com.example.myapplication.R
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import androidx.navigation.Navigation
import com.squareup.picasso.Picasso
import java.lang.Exception
import java.util.ArrayList

class CatalogAdapter : RecyclerView.Adapter<CatalogAdapter.MyView> {
    // List with String type
    private var Products: ArrayList<Product>? = null

    // View Holder class which
    // extends RecyclerView.ViewHolder
    inner class MyView(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
        // Text View
        var CatalogItemName: TextView
        var CatalogPicture: ImageView
        var CatalogItemPrice: TextView
        override fun onClick(view: View) {
            val item = Products!![adapterPosition]
            val bundle = Bundle()
            bundle.putString("id", item.productId.toString())
            try {
                Navigation.findNavController(view).navigate(R.id.productFragment, bundle)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }

        // parameterised constructor for View Holder class
        // which takes the view as a parameter
        init {
            view.setOnClickListener(this)
            // initialise TextView with id
            CatalogItemName = view.findViewById<View>(R.id.CatalogItemName) as TextView
            CatalogPicture = view.findViewById<View>(R.id.CatalogPicture) as ImageView
            CatalogItemPrice = view.findViewById<View>(R.id.CatalogItemPrice) as TextView
        }
    }

    // Constructor for adapter class
    // which takes a list of String type
    constructor(horizontalList: ArrayList<Product>) {
        Products = horizontalList
    }

    constructor() {}

    fun updateUserList(userArrayList: ArrayList<Product>?) {
        Products!!.clear()
        Products = userArrayList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): MyView {
        val itemView = LayoutInflater
            .from(parent.context)
            .inflate(
                R.layout.catalogitem,
                parent,
                false
            )
        return MyView(itemView)
    }

    override fun onBindViewHolder(holder: MyView, position: Int) {

        // Привязка конкретного элемента
        try {
            holder.CatalogItemName.text = Products!![position].name
            val price = Products!![position].price.toString()
            if (Products!![position].count <= 0) {
                holder.CatalogItemPrice.setText(R.string.NotInStock)
            } else holder.CatalogItemPrice.text = price + "P"
        } catch (e: Exception) {
            e.printStackTrace()
        }
        val url = Products!![position].photo
        try {
            Picasso.get() // Context
                .load("http://192.168.1.180:82$url") // URL or file
                .into(holder.CatalogPicture)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun getItemCount(): Int {
        return Products!!.size
    }
}